feat(readme): add file transfer readme (#170)
edgehog-device-manager/edgehog-zephyr-device: samples/edgehog_app/README.md

## Before you begin

Each Edgehog device requires an Edgehog instance to connect to, which itself relies on an Astarte instance.

> You could setup Edgehog following [this guide](https://docs.edgehog.io/0.12/edgehog_in_5_minutes.html) or you could use an [Astarte Cloud](https://astarte.cloud/) instance.

Reviewer: ```suggestion
You could setup Edgehog following [this guide](https://docs.edgehog.io/snapshot/edgehog_in_5_minutes.html) or you could use an [Astarte Cloud](https://astarte.cloud/) instance.
```
Not 100% sure this my suggestion is an improvement. What do you think?
Snapshot points at `master` so it's not always perfect.

Author: Since master could be unstable, probably it's better to use the latest pinned version

Depending on the type of Astarte/Edgehog instances you are using the configuration parameters for
this sample will be slightly different.

If you are usign a managed Astarte/Edgehog instance such as `astarte.cloud` you will be able to fully take advantage of the secure TLS connection of Astarte.

Set the following configuration entries in the `prj.conf` file of this sample.
```conf
CONFIG_ASTARTE_DEVICE_SDK_HOSTNAME="<HOSTNAME>"
CONFIG_ASTARTE_DEVICE_SDK_HTTPS_CA_CERT_TAG=1
CONFIG_ASTARTE_DEVICE_SDK_MQTTS_CA_CERT_TAG=1
CONFIG_ASTARTE_DEVICE_SDK_CLIENT_CERT_TAG=2
CONFIG_ASTARTE_DEVICE_SDK_REALM_NAME="<REALM_NAME>"

CONFIG_ASTARTE_DEVICE_ID="<DEVICE_ID>"
CONFIG_ASTARTE_CREDENTIAL_SECRET="<CREDENTIAL_SECRET>"
```
Where `<DEVICE_ID>` is the device ID of the device you would like to use in the sample, `<HOSTNAME>` is the hostname for your Astarte instance, `<REALM_NAME>` is the name of your testing realm and `<CREDENTIAL_SECRET>` is the credential secret obtained through the manual registration.

In addition, the file `ca_certificates.h` should be modified, placing in the `astarte_ca_certificate_root` array a valid CA certificate in the PEM format.

### Transport layer choice

> Depending on which board you use different transport layer may be available to you. This sample supports configurations for Ethernet and WiFi.

Reviewer: ```suggestion
Depending on which board you use, different transport layers may be available to you. This sample supports configurations for Ethernet and WiFi.
```

### Over the air (OTA) updates

A central part of Edgehog is that it provides over the air updates for large device fleets. An [OTA Updates](../../doc/architecture/Over%20The%20Air%20(OTA)%20updates.md) is a method for delivering firmware updates to remote devices using a network connection. This sample will show you how to build and deploy a device using `sysbuild` that leverages `MCUboot` capabilities offering out of the box OTA updates.

### Sysbuild specific settings

Building with --sysbuild automatically includes and configures MCUboot to enable OTA updates.

Build the sample using:

```sh
west build -b stm32h573i_dk --sysbuild samples/edgehog_app
```

> The board overlay for MCUboot is saved in the `sysbuild/boards/` directory. To support new boards, simply add the corresponding overlay file to this folder and it will be loaded automatically. Alternatively, you can pass the overlay manually using `-Dmcuboot_DTC_OVERLAY_FILE=/path/to/overlay`.

Reviewer: ```suggestion
The board overlay for MCUboot is searched in the `sysbuild/boards/` directory. To support new boards, simply add the corresponding overlay file to this folder and it will be loaded automatically. Alternatively, you can pass the overlay manually using `-Dmcuboot_DTC_OVERLAY_FILE=/path/to/overlay`.
```

The sample implements a **loopback mechanism** leveraging in-memory `k_pipe` objects:
- When a download transfer is initiated with the filename `loopback`, the sample intercepts the stream via `.on_stream_transfer_start` and spawns a dedicated download thread. The payload is written directly to a RAM buffer dynamically.
- When an upload transfer is requested matching the `loopback` name, an upload thread is spawned that pushes the previously buffered payload back up to the server.

This mechanism allows you to test both incoming and outgoing data flows rapidly without requiring persistent storage (flash/filesystem) configured on your target board. For file system operations, the sample currently logs when an `.on_filesystem_transfer_done` event occurs.

To enable this, make sure to wire the provided file transfer callbacks to your device configuration in `main.c` before starting the Edgehog device:

```C
edgehog_device_config_t edgehog_conf = {
    // ... other configurations ...
    .file_transfer_cbks = {
        .on_stream_transfer_start = app_on_stream_transfer_start,
        .on_filesystem_transfer_done = app_on_filesystem_transfer_done
    }

> };

Reviewer: This is already done by default in the sample right? 
I think we can avoid mentioning it.

Author: Yes
